### DISCUSSION TITLE
Button Link Dynamic URL: Allow dynamic editing of field content with functions like lowercase()

### DISCUSSION THREAD

**Discussion by philippdzm** [directus/directus#18902] [Feature Requests]
### Summary

The button link feature is very helpful, especially with its dynamic URL builder! I love it.
However, when using the dynamic URL feature, including fields may need the use of functions like `lowercase()`.

In my application that's the case. I assume there are other functions which could be helpful, here are some other ideas:

- `lowercase()`
- `uppercase()`
- `encodeURIComponent()`
- `replace()`: This function would take two parameters, the string to find and the string to replace it with.
- `removeSymbols()`: This function could remove all non-alphanumeric characters from a string.
- `regexReplace()`: This function would take two parameters, the pattern to find and the string to replace it with.

### Basic Example

`example.com/{{field_1}}`

data in field_1 may be: `ABCDEF`

link: `example.com/ABCDEF` instead of `example.com/abcdef`

### Motivation

Instead of having to manually enter the URL with each entry, the URL could be automatically generated. Users would be uber happy.

### Detailed Design

`example.com/{{field_1.lowercase()}}`

NB: I am not sure how to show best the application of that function.

### Must Have:
- `lowercase()`
- `uppercase()`

### Should Have:
- `removeSymbols()`

### Could Have:
- `encodeURIComponent()`
- `replace()`
- `replaceRegex()`

### Drawbacks

- note sure about the drawbacks

### Alternatives

none

### Adoption Strategy

Pretty easy: The new button link can be created, tested and old columns containing can be removed next.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
